app/controllers/trainees/lead_schools_controller.rb

```diff
@@ -11,10 +11,13 @@ def index
 
     def update
       @trainee_lead_school_form = TraineeLeadSchoolForm.new(trainee, params: trainee_params, user: current_user)
+
+      return redirect_to trainee_lead_schools_path(@trainee, query: query) if searching_again?
```

Comment:
I think we need to add some kind of validation to the search again field within the `TraineeLeadSchoolForm`, if you check the radio button but don't enter any text then this `searching_again?` check evaluates to false and it tries to save the school (there isn't one), leading to this error:

```
PG::ForeignKeyViolation: ERROR: insert or update on table "trainees" violates foreign key constraint "fk_rails_6b8c2e6680" 
DETAIL: Key (lead_school_id)=(0) is not present in table "schools".
```

Reply:
Or I can just replace `params[:search_again].present?` with `params.key?(:search_again)` which fixes the issue where the query is a blank string.

Reply:
Ed suggested we validate for empty searches, so I'll add that in now.